src/commands/info/lookup.ts

```diff
@@ -24,36 +25,53 @@ class LookupCommand extends Command
 		});
 	}
 
-	public parseArgs(message: GuildMessage, [code]: string[]): string | Promise<string | Invite[]>
+	public async parseArgs(message: GuildMessage, [code]: string[]): Promise<string | [Invite]>
 	{
 		if (!code) return 'you have to give me an invite link or code!';
 
-		return this.client.fetchInvite(code)
-			.then((invite: Invite) => ([invite]))
-			.catch(() => 'I couldn\'t find a valid invite for this link or code.');
+		try
+		{
+			const invite: Invite = await this.client.fetchInvite(code);
+
+			if (!invite.guild) return 'this invite is not for a guild.';
```

Comment:
I think `this invite is invalid!` would be more fitting than `this invite is not for a guild.`

Reply:
That would be misleading; the invite _is_ valid. It's just not an invite leading to a guild.

Reply:
Alright I'll approve but, how would an invite be valid and not lead to a guild? Where else would it lead to?

Reply:
Group DMs

Reply:
Oh yeah, that's a thing now. Good catch.